samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend fails on qemu_riscv32_smp

**Describe the bug**
The `samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend` test suite fails to compile on `qemu_riscv32_smp` following commit 9470e3d129a364f280285293c7e4034de833978c.

**To Reproduce**
Steps to reproduce the behavior:
1. `twister -v -p qemu_riscv32_smp --test samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend`
2. See error:

```
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /home/brix/Projects/zephyrproject/zephyr/twister-out/qemu_riscv32_smp/samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend/zephyr/zephyr_pre0.map 
: && ccache /opt/zephyr-sdk/0.16.1/riscv64-zephyr-elf/bin/riscv64-zephyr-elf-gcc  -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/riscv/core/offsets/offsets.c.obj  -fuse-ld=bfd  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/home/brix/Projects/zephyrproject/zephyr/twister-out/qemu_riscv32_smp/samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/riscv/core/libarch__riscv__core.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L"/opt/zephyr-sdk/0.16.1/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/rv32imac_zicsr_zifencei/ilp32"  -L/home/brix/Projects/zephyrproject/zephyr/twister-out/qemu_riscv32_smp/samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend/zephyr  -lgcc  zephyr/arch/common/libisr_tables.a  -Wl,--fatal-warnings  -mabi=ilp32  -march=rv32imac_zicsr_zifencei  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -DPICOLIBC_INTEGER_PRINTF_SCANF  --specs=picolibc.specs  -lc  -lgcc && cd /home/brix/Projects/zephyrproject/zephyr/twister-out/qemu_riscv32_smp/samples/subsys/logging/dictionary/sample.logger.basic.dictionary.uart_int_frontend/zephyr && /home/brix/Projects/zephyrproject/venv/lib/python3.10/site-packages/cmake/data/bin/cmake -E true
/opt/zephyr-sdk/0.16.1/riscv64-zephyr-elf/bin/../lib/gcc/riscv64-zephyr-elf/12.2.0/../../../../riscv64-zephyr-elf/bin/ld.bfd:zephyr/linker_zephyr_pre0.cmd:120: warning: memory region `DEVNULL_REGION' not declared
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Expected behavior**
Build succeeds.

**Impact**
This affects CI. See e.g. https://github.com/zephyrproject-rtos/zephyr/actions/runs/6698096228/job/18199444207

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: 7db93097daaa8a843529c2c4f97e5c97736c8e12

--- a/include/zephyr/linker/linker-devnull.h
+++ b/include/zephyr/linker/linker-devnull.h
@@ -65,13 +65,9 @@
 #error "Cannot place devnull segment adjacent to ROM region."
 #endif
 
-#if defined(CONFIG_LINKER_DEVNULL_MEMORY)
-#define DEVNULL_REGION DEVNULL_ROM
-#else
-#define DEVNULL_REGION ROMABLE_REGION
-#endif
-
 #endif /* CONFIG_XIP */
+
+#define DEVNULL_REGION DEVNULL_ROM
 
 #endif /* CONFIG_LINKER_DEVNULL_MEMORY */
 

